jis-api/src/main/java/pl/edu/uj/JImageStream/api/core/Filter.java

```diff
@@ -39,14 +39,8 @@ protected void setPixel(int x, int y, Pixel pixel) {
             colorRestriction.process(sourceColors, outputColors);
         }
 
-        //todo needs refactoring
         for(int i = 0; i < sourceColors.length; ++i){
-            if(sourceColors[i] > 255){
-                sourceColors[i] = 255;
-            }
-            if(sourceColors[i] < 0){
-                sourceColors[i] = 0;
-            }
+            sourceColors[i] = Math.min(Math.max(sourceColors[i], 0), 255);
         }
 
         output.setPixel(x, y, sourceColors);
```

Comment:
@" //todo alpha support, investigate when and where alpha chanel is present" havent we already done that?